src/charm.py

```diff
@@ -319,10 +329,46 @@ def _update_ingress(self) -> None:
             ingress_controller = self._get_ingress_controller(namespace)
             ingress_ips = ingress_controller.get_ingress_ips()
             message = f"Ingress IP(s): {', '.join(ingress_ips)}" if ingress_ips else ""
+
+            if self.model.get_relation("ingress") is not None:
+                url = self._generate_ingress_url(hostnames) or ingress_ips[0]
```

Comment:
`ingress_ips[0]` won't include the scheme or path, but it should (we should also have a unit test that confirms the case when we don't have a hostname).